### PR TITLE
Standardize naming in WellKnownTypeNames

### PR DESCRIPTION
**Pull request by Evangelink**
Add suffix to the names matching the number of generic type arguments.

Also fix some broken ordering in the constants.

### CODE REVIEW

**Comment by Evangelink** [comment]
@mavasani It seems that there are some types not present in the 2.9.x, I will keep this PR and issue a new one for the 2.9.x branch. Let me know if you prefer not to keep this one.

**Comment by dotpaul** [comment]
I merged the 2.9.x changes into the master branch. Lemme know if you see something wrong (or just fix it :smiley:).

**Comment by Evangelink** [comment]
@dotpaul LGTM, would you mind reviewing this PR if you have some free time?

File: src/Utilities/Compiler/WellKnownTypeNames.cs [hunk 1]
```diff
@@ -296,16 +296,17 @@ internal static class WellKnownTypeNames
         public const string SystemThreadingInterlocked = "System.Threading.Interlocked";
         public const string SystemThreadingMonitor = "System.Threading.Monitor";
         public const string SystemThreadingSpinLock = "System.Threading.SpinLock";
-        public const string SystemThreadingTasksTask1 = "System.Threading.Tasks.Task`1";
-        public const string SystemThreadingTasksTask = "System.Threading.Tasks.Task";
+        public const string SystemThreadingTasksGenericTaskCompletionSource = "System.Threading.Tasks.TaskCompletionSource`1";
```
Comment (dotpaul) [inline src/Utilities/Compiler/WellKnownTypeNames.cs:299]: I think you meant to get rid of these two.

Comment (dotpaul) [reply src/Utilities/Compiler/WellKnownTypeNames.cs:299]: Or get rid of SystemThreadingTasksGenericValueTask, and rename SystemThreadingTasksGenericTaskCompletionSource  to SystemThreadingTasksTaskCompletionSource1.

Comment (Evangelink) [reply src/Utilities/Compiler/WellKnownTypeNames.cs]: Thank you, I missed it when I did the conflict merge.

File: src/Utilities/Compiler/WellKnownTypeNames.cs [hunk 2]
```diff
@@ -296,16 +296,16 @@ internal static class WellKnownTypeNames
         public const string SystemThreadingInterlocked = "System.Threading.Interlocked";
         public const string SystemThreadingMonitor = "System.Threading.Monitor";
         public const string SystemThreadingSpinLock = "System.Threading.SpinLock";
-        public const string SystemThreadingTasksTask1 = "System.Threading.Tasks.Task`1";
+        public const string SystemThreadingTasksGenericTaskCompletionSource1 = "System.Threading.Tasks.TaskCompletionSource`1";
```
Comment (dotpaul) [inline src/Utilities/Compiler/WellKnownTypeNames.cs:299]:
```suggestion
        public const string SystemThreadingTasksTaskCompletionSource1 = "System.Threading.Tasks.TaskCompletionSource`1";
```

Comment (Evangelink) [reply src/Utilities/Compiler/WellKnownTypeNames.cs:299]: Oh crap... Sorry for the wasted time!

Comment (dotpaul) [reply src/Utilities/Compiler/WellKnownTypeNames.cs]: No worries, @Evangelink ! :smiley: Sorry I didn't mention to also rename the references.

**Comment by Evangelink** [comment]
@dotpaul Now that shall be ok ok ;)

**Pull request merged**